python_modules/dagster/dagster/_core/definitions/sensor_definition.py

```diff
@@ -153,6 +155,107 @@ def update_cursor(self, cursor: Optional[str]) -> None:
+    """The context object available as the argument to the evaluation function of a :py:class:`dagster.MultiAssetSensorDefinition`.
+
+    Users should not instantiate this object directly. To construct a
+    `MultiAssetSensorEvaluationContext` for testing purposes, use :py:func:`dagster.
+    build_multi_asset_sensor_context`.
+
+    Attributes:
+        instance_ref (Optional[InstanceRef]): The serialized instance configured to run the schedule
+        cursor (Optional[str]): The cursor, passed back from the last sensor evaluation via
+            the cursor attribute of SkipReason and RunRequest
+        last_completion_time (float): DEPRECATED The last time that the sensor was evaluated (UTC).
+        last_run_key (str): DEPRECATED The run key of the RunRequest most recently created by this
+            sensor. Use the preferred `cursor` attribute instead.
+        repository_name (Optional[str]): The name of the repository that the sensor belongs to.
+        instance (Optional[DagsterInstance]): The deserialized instance can also be passed in
+            directly (primarily useful in testing contexts).
+
+    Example:
+
+    .. code-block:: python
+
+        from dagster import sensor, SensorEvaluationContext
+
+        @sensor
+        def the_sensor(context: SensorEvaluationContext):
+            ...
+
+    """
+
+    def get_latest_events(self, asset_keys: Sequence[AssetKey]):
+        from dagster._core.events import DagsterEventType
+        from dagster._core.storage.event_log.base import EventRecordsFilter
+
+        cursor_dict = json.loads(self.cursor) if self.cursor else {}
+        asset_event_records = {}
+        for a in asset_keys:
+            event_records = self.instance.get_event_records(
+                EventRecordsFilter(
+                    event_type=DagsterEventType.ASSET_MATERIALIZATION,
+                    asset_key=a,
+                    after_cursor=cursor_dict.get(a.to_user_string()),
+                ),
+                ascending=False,
+                limit=1,
+            )
+
+            if event_records:
+                asset_event_records[a.to_user_string()] = event_records[0].event_log_entry
+            else:
+                asset_event_records[a.to_user_string()] = None
+
+        return asset_event_records
+
+    def get_all_events(self, asset_keys: Sequence[AssetKey]):
+        from dagster._core.events import DagsterEventType
+        from dagster._core.storage.event_log.base import EventRecordsFilter
+
+        cursor_dict = json.loads(self.cursor) if self.cursor else {}
+        asset_event_records = {}
+        for a in asset_keys:
+            event_records = self.instance.get_event_records(
+                EventRecordsFilter(
+                    event_type=DagsterEventType.ASSET_MATERIALIZATION,
+                    asset_key=a,
+                    after_cursor=cursor_dict.get(a.to_user_string()),
+                ),
+                ascending=False,
```

Comment:
flip to True so that the latest events are first (to support if we want to process for each event)

Reply:
I think I would expect the latest events to come last

Reply:
yep, typo in my comment oops